### PR TITLE
Added two variants of ComplexVis: phase map and amplitude map

### PR DESCRIPTION
**Pull request by loichuder**
Continuation of work on #310 

image

Also changed the aspect of imaginary `i` when displaying complex data:
image
image

### CODE REVIEW

File: src/h5web/vis-packs/core/complex/MappedComplexVis.tsx [hunk 1]
```diff
@@ -62,18 +66,18 @@ function MappedPhaseMapVis(props: Props) {
...
+  const dataDomain =
+    visType !== ComplexVisType.Amplitude ? phaseDomain : amplitudeDomain;
 
+  const visDomain = useVisDomain(customDomain, dataDomain);
+  const [safeDomain] = useSafeDomain(visDomain, dataDomain, scaleType);
   useEffect(() => {
-    setDataDomain(phaseDomain);
-  }, [phaseDomain, setDataDomain]);
+    setDataDomain(dataDomain);
+  }, [dataDomain, setDataDomain, visType, scaleType]);
```
Comment (loichuder) [inline src/h5web/vis-packs/core/complex/MappedComplexVis.tsx:80]: `dataDomain` needs to be reset when changing the `visType` or the `scaleType` so I added these deps to the `useEffect`.

File: src/h5web/vis-packs/core/matrix/MappedMatrixVis.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ function MappedMatrixVis(props: Props) {
       <MatrixVis
         dataArray={mappedArray}
         formatter={(dataValue) => renderComplex(dataValue, '.2e')}
-        cellWidth={216} // To accommodate the longer complex numbers
+        cellWidth={232} // To accommodate the longer complex numbers
```
Comment (loichuder) [inline src/h5web/vis-packs/core/matrix/MappedMatrixVis.tsx:38]: Making a bit more space for the new ⅈ

File: src/h5web/vis-packs/core/visualizations.ts [hunk 3]
```diff
@@ -17,19 +17,21 @@ import {
...
+import { ComplexConfigProvider } from './complex/config';
 
 export enum Vis {
   Raw = 'Raw',
   Scalar = 'Scalar',
   Matrix = 'Matrix',
   Line = 'Line',
   Heatmap = 'Heatmap',
-  PhaseMap = 'Phase map',
+  Complex = 'Complex',
```
Comment (loichuder) [inline src/h5web/vis-packs/core/visualizations.ts:34]: I am not awfully happy with the name **Complex** so I am open to suggestions

File: src/h5web/vis-packs/core/heatmap/config.tsx [hunk 4]
```diff
@@ -87,14 +87,14 @@ export function HeatmapConfigProvider(props: ConfigProviderProps) {
   const { children } = props;
 
   // https://github.com/pmndrs/zustand/issues/346
-  const [reydrated, setRehydrated] = useState(false);
+  const [rehydrated, setRehydrated] = useState(false);
```
Comment (loichuder) [inline src/h5web/vis-packs/core/heatmap/config.tsx:90]: Typo correction

File: src/h5web/toolbar/ComplexToolbar.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    setColorMap,
+    scaleType,
+    setScaleType,
+    keepAspectRatio,
+    toggleAspectRatio,
+    showGrid,
+    toggleGrid,
+    invertColorMap,
+    toggleColorMapInversion,
+  } = useHeatmapConfig((state) => state, shallow);
```
Comment (loichuder) [inline src/h5web/toolbar/ComplexToolbar.tsx:33]: For now, the `ComplexToolbar` uses the same config as the `HeatmapToolbar` which means setting from the `Heatmap` will be persisted in the `Complex` and vice-versa.

We will wait for user feedback to see if this should be changed

File: src/h5web/vis-packs/core/complex/MappedComplexVis.tsx [hunk 6]
```diff
@@ -83,11 +87,19 @@ function MappedPhaseMapVis(props: Props) {
...
-      title={title}
+      dataArray={dataArray}
+      domain={safeDomain}
+      alphaArray={
+        visType === ComplexVisType.PhaseAndAmplitude
+          ? amplitudeArray
+          : undefined
+      }
+      alphaDomain={
+        visType === ComplexVisType.PhaseAndAmplitude
```
Comment (axelboc) [inline src/h5web/vis-packs/core/complex/MappedComplexVis.tsx:98]: Could you maybe renamed `PhaseAndAmplitude` to just `PhaseAmplitude`? The `And` seems unnecessary.

File: src/h5web/vis-packs/core/complex/MappedComplexVis.tsx [hunk 7]
```diff
@@ -83,11 +87,19 @@ function MappedPhaseMapVis(props: Props) {
...
+          ? amplitudeDomain
+          : undefined
+      }
+      title={`${title || ''} (${visType.toLowerCase()})`}
```
Comment (axelboc) [inline src/h5web/vis-packs/core/complex/MappedComplexVis.tsx:102]: Seems to me like this logic should be a bit more complex: if no `title` is present, there's a leading space and the vis type starts with a lower case.

File: src/h5web/vis-packs/core/visualizations.ts (hunk 3 above)
Comment (axelboc) [reply src/h5web/vis-packs/core/visualizations.ts:34]: I don't mind it, but we'll ask Thomas and Andy on Tuesday 😉

**Review by axelboc — APPROVED** [review]
Nice !

**Pull request merged**